Group reflection context docs together and provide new topic on behavior context (#1360)
o3de/o3de.org: content/docs/user-guide/visualization/animation/character-editor/custom-events-parameters-creating-eventdata-types.md

>    For more information about the `Equal` function, see [More About the Equal Function](#char-animation-editor-custom-events-parameters-the-equal-function).

Reviewer: ```suggestion
For more information about the `Equal` function, refer to [More About the Equal Function](#more-about-the-equal-function).
```

> 1. Implement the `Reflect` method to reflect the type to the [Serialization Context](/docs/user-guide/programming/components/reflection/serialization-context/) and [Edit Context](/docs/user-guide/programming/components/reflection/edit-context/) contexts.

Reviewer: ```suggestion
1. Implement the `Reflect` method to reflect the type to the [serialization context](/docs/user-guide/programming/components/reflection/serialization-context/) and [edit context](/docs/user-guide/programming/components/reflection/edit-context/).
```

>    When you reflect the event to the edit context, add the `Creatable` attribute to `ClassElement`. This makes the `EventData` type visible in the Animation Editor's **Motion Events** tab so that users can select it.

Reviewer: ```suggestion
When you reflect the event to the edit context, add the `Creatable` attribute to `ClassElement`. This makes the `EventData` type visible in **Animation Editor** in the **Motion Events** tab so that users can select it.
```
